transport/http/src/test/java/org/kaazing/gateway/transport/http/HttpAcceptorRule.java

```diff
@@ -0,0 +1,107 @@
+import org.kaazing.gateway.transport.nio.internal.NioSocketAcceptor;
+import org.kaazing.gateway.transport.nio.internal.NioSocketConnector;
+import org.kaazing.gateway.util.scheduler.SchedulerProvider;
+
+import java.net.URI;
+import java.util.Collections;
+
+
+/**
+ * Declaring an instance of this class as a @Rule causes the gateway to be
+ * started in process before each test method and stopped after it. The rule
+ * can be chained with a K3poRule for use with robot (this causes Robot to be
+ * started before the gateway and stopped after it).
+ */
+public class HttpAcceptorRule implements TestRule {
+
+    private ResourceAddressFactory addressFactory;
+    private HttpAcceptor httpAcceptor;
+
+    @Override
+    public Statement apply(Statement base, Description description) {
+        return new AcceptorStatement(base);
+    }
+
+    public void bind(String address, IoHandler acceptHandler) {
+        ResourceAddress resourceAddress = addressFactory.newResourceAddress(URI.create(address));
+        bind(resourceAddress, acceptHandler);
+    }
+
+    public void bind(ResourceAddress acceptAddress, IoHandler acceptHandler) {
+        httpAcceptor.bind(acceptAddress, acceptHandler, null);
+    }
+
+    private final class AcceptorStatement extends Statement {
+
+        private final Statement base;
+
+        private NioSocketConnector tcpConnector;
+        private NioSocketAcceptor tcpAcceptor;
+        private SchedulerProvider schedulerProvider;
+
+        public AcceptorStatement(Statement base) {
+            this.base = base;
+        }
+
+        @Override
+        public void evaluate() throws Throwable {
+            try {
+                schedulerProvider = new SchedulerProvider();
+
+                addressFactory = ResourceAddressFactory.newResourceAddressFactory();
+                TransportFactory transportFactory = TransportFactory.newTransportFactory(Collections.<String, Object> emptyMap());
+                BridgeServiceFactory serviceFactory = new BridgeServiceFactory(transportFactory);
+
+                tcpAcceptor = (NioSocketAcceptor)transportFactory.getTransport("tcp").getAcceptor();
+                tcpAcceptor.setResourceAddressFactory(addressFactory);
+                tcpAcceptor.setBridgeServiceFactory(serviceFactory);
+                tcpAcceptor.setSchedulerProvider(schedulerProvider);
+
+                tcpConnector = (NioSocketConnector)transportFactory.getTransport("tcp").getConnector();
+                tcpConnector.setResourceAddressFactory(addressFactory);
+                tcpConnector.setBridgeServiceFactory(serviceFactory);
+                tcpConnector.setTcpAcceptor(tcpAcceptor);
+
+                httpAcceptor = (HttpAcceptor)transportFactory.getTransport("http").getAcceptor();
+                httpAcceptor.setBridgeServiceFactory(serviceFactory);
+                httpAcceptor.setResourceAddressFactory(addressFactory);
+                httpAcceptor.setSchedulerProvider(schedulerProvider);
```

Comment:
You can use transportFactory.injectResources to simplify the code and avoid having to create the tcpAcceptor and tcpConnector. That method automatically injects resources which are connectors or acceptors:
```
                schedulerProvider = new SchedulerProvider();
                addressFactory = ResourceAddressFactory.newResourceAddressFactory();
                TransportFactory transportFactory = TransportFactory.newTransportFactory(Collections.<String, Object> emptyMap());

                Map<String, Object> resources = new HashMap<String, Object>();
                resources.put("schedulerProvider",  schedulerProvider);
                resources.put("configuration",  new Properties());
                resources.put("bridgeServiceFactory", new BridgeServiceFactory(transportFactory));
                resources.put("resourceAddressFactory", addressFactory);
                transportFactory.injectResources(resources);

                httpAcceptor = (WsnAcceptor)transportFactory.getTransport("http").getAcceptor();

                base.evaluate();
```